packages/server/src/integrations/utils.ts

```diff
@@ -295,42 +292,41 @@ function copyExistingPropsOver(
 
 /**
  * Look through the final table definitions to see if anything needs to be
- * copied over from the old and if any errors have occurred mark them so
- * that the user can be made aware.
+ * copied over from the old.
  * @param tables The list of tables that have been retrieved from the external database.
  * @param entities The old list of tables, if there was any to look for definitions in.
  */
 export function finaliseExternalTables(
-  tables: { [key: string]: any },
-  entities: { [key: string]: any }
-) {
-  const invalidColumns = Object.values(InvalidColumns)
-  let finalTables: { [key: string]: any } = {}
-  const errors: { [key: string]: string } = {}
-  // @ts-ignore
-  const tableIds: [string] = Object.values(tables).map(table => table._id)
+  tables: Record<string, ExternalTable>,
+  entities: Record<string, ExternalTable>
+): Record<string, ExternalTable> {
+  let finalTables: Record<string, Table> = {}
+  const tableIds = Object.values(tables).map(table => table._id!)
   for (let [name, table] of Object.entries(tables)) {
-    const schemaFields = Object.keys(table.schema)
-    // make sure every table has a key
-    if (table.primary == null || table.primary.length === 0) {
-      errors[name] = BuildSchemaErrors.NO_KEY
-      continue
-    } else if (
-      schemaFields.find(field =>
-        invalidColumns.includes(field as InvalidColumns)
-      )
-    ) {
-      errors[name] = BuildSchemaErrors.INVALID_COLUMN
-      continue
-    }
-    // make sure all previous props have been added back
     finalTables[name] = copyExistingPropsOver(name, table, entities, tableIds)
   }
-  // sort the tables by name
-  finalTables = Object.entries(finalTables)
+  // sort the tables by name (TODO(samwho): why?)
+  return Object.entries(finalTables)
```

Comment:
on the TODO - really its just for a consistent response from the API, you can expect the tables to come back in a predictable order.